test/02-device-types.ts

```diff
@@ -1,7 +1,64 @@
+import { permissions } from '@balena/pinejs';
+import { api } from '@balena/pinejs/out/sbvr-api/sbvr-utils';
 import * as _ from 'lodash';
 import { expect } from './test-lib/chai';
 
 import { supertest } from './test-lib/supertest';
+// All of these test device types are not part of the contracts, so we have to include them manually until we stop syncing with S3.
+export const addFakeDeviceTypes = async () => {
+	await Promise.all(
+		[
+			{
+				id: 991,
+				slug: 'dt-with-ignored-release',
+				name: 'dt-with-ignored-release',
+			},
+			{
+				id: 992,
+				slug: 'dt-with-403-ignore-file-release',
+				name: 'dt-with-403-ignore-file-release',
+			},
+			{
+				id: 993,
+				slug: 'dt-with-empty-device-type-json-release',
+				name: 'dt-with-empty-device-type-json-release',
+			},
+			{
+				id: 994,
+				slug: 'dt-with-404-device-type-json-release',
+				name: 'dt-with-404-device-type-json-release',
+			},
+			{
+				id: 995,
+				slug: 'dt-with-500-ignore-file-release',
+				name: 'dt-with-500-ignore-file-release',
+			},
+			{
+				id: 996,
+				slug: 'dt-with-500-device-type-json-release',
+				name: 'dt-with-500-device-type-json-release',
+			},
+			{
+				id: 997,
+				slug: 'dt-with-no-valid-releases',
+				name: 'dt-with-no-valid-releases',
+			},
+			{
+				id: 998,
+				slug: 'dt-with-failing-listing',
+				name: 'dt-with-failing-listing',
+			},
+		].map(async (dt) => {
+			return await api.resin.post({
+				resource: 'device_type',
+				passthrough: {
+					req: permissions.root,
+				},
+				body: dt,
```

Comment:
Minor performance optimization since we don't care about the result:
```suggestion
				passthrough: {
					req: permissions.root,
				},
				options: { returnResource: false },
				body: dt,
```